src/components/organize/CampaignDetailsHeader.tsx

```diff
@@ -0,0 +1,156 @@
+import NextLink from 'next/link';
+import { useRouter } from 'next/router';
+import { useState } from 'react';
+import { Avatar, Box, Button, makeStyles, Typography } from '@material-ui/core';
+import { FormattedDate, FormattedMessage as Msg, useIntl } from 'react-intl';
+import { useMutation, useQuery, useQueryClient } from 'react-query';
+
+import CampaignForm from '../CampaignForm';
+import ZetkinDialog from '../ZetkinDialog';
+
+import getCampaignEvents from '../../fetching/getCampaignEvents';
+import patchCampaign from '../../fetching/patchCampaign';
+import { ZetkinCampaign } from '../../types/zetkin';
+
+const useStyles = makeStyles((theme) => ({
+    personContainer: {
+        '&:hover': {
+            cursor: 'pointer',
+            textDecoration: 'underline',
+        },
+    },
+    responsiveFlexBox: {
+        display: 'flex',
+        [theme.breakpoints.down('sm')]: {
+            flexDirection: 'column',
+        },
+    },
+    responsiveText: {
+        width: '70%',
+        [theme.breakpoints.down('sm')]: {
+            width: '100%',
+        },
+    },
+}));
+
+interface CampaignDetailsHeaderProps {
+    campaign: ZetkinCampaign;
+}
+
+const CampaignDetailsHeader: React.FunctionComponent<CampaignDetailsHeaderProps> = ({ campaign }) => {
+    const intl = useIntl();
+    const queryClient = useQueryClient();
+    const [editCampaignDialogOpen, setEditCampaignDialogOpen] = useState(false);
+    const { campId, orgId } = useRouter().query;
+    const classes = useStyles();
+
+    const { data: events } = useQuery(
+        ['campaignEvents', orgId, campId],
+        getCampaignEvents(orgId as string, campId as string),
+    );
+
+    const sortedEvents = events ? [...events].sort((a, b) => {
+        return new Date(a.start_time).getTime() - new Date(b.start_time).getTime();
+    }): [];
+
+    const startDate = sortedEvents[0]?.start_time;
+    const endDate = sortedEvents[sortedEvents.length - 1]?.start_time;
+
+    const campaignMutation = useMutation(patchCampaign(orgId as string, campId as string), {
+        onSettled: () => queryClient.invalidateQueries('campaign'),
+    });
+
+    const handleEditCampaignFormSubmit = (data: Record<string, unknown>) => {
+        campaignMutation.mutate(data);
+        setEditCampaignDialogOpen(false);
+    };
+
+    return (
+        <>
+            <Box p={ 4 }>
+                { /* Description */ }
+                { campaign.info_text && (
+                    <Typography variant="body1">
```

Comment:
I think this should have a max-width (or equivalent), so that it does not render super-wide on wide screens. You know better than I do how best to solve this in MUI. 😊 

This screen I think shows the legibility issue with the current layout:

![image](https://user-images.githubusercontent.com/550212/126120617-e5fbb873-45ea-4f47-8920-ead6585813f6.png)

Reply:
that makes sense, i can do that :+1: